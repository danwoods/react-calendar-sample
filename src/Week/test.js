import React from 'react'
import { Week } from './index.js'
import renderer from 'react-test-renderer'

const date = new Date(2000, 0, 1)
const monthDate = new Date(2000, 1, 1)
const Day = () => 'day comp'

describe('Week component', () => {
  test('Basic Week', () => {
    const component = renderer.create(
      <Week date={date} displayForMonth={monthDate} />
    )
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Week with Day component', () => {
    const component = renderer.create(
      <Week date={date} displayForMonth={monthDate} DayComponent={Day} />
    )
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Week with full props', () => {
    const component = renderer.create(
      <Week
        date={date}
        displayForMonth={monthDate}
        DayComponent={Day}
        rootClassName={'rcn'}
      />
    )
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
