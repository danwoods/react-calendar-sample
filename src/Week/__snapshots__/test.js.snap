// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Week component Basic Week 1`] = `
<div
  className="TEST-week-root "
  data-component-id="week"
>
  <div
    className="TEST-day-root "
    title="December 26, 1999"
  >
    <div
      className="TEST-day-content "
    >
      <h6
        className="TEST-day-date-label TEST-different-month"
      >
        26
      </h6>
    </div>
  </div>
  <div
    className="TEST-day-root TEST-week-day-root"
    title="December 27, 1999"
  >
    <div
      className="TEST-day-content "
    >
      <h6
        className="TEST-day-date-label TEST-different-month"
      >
        27
      </h6>
    </div>
  </div>
  <div
    className="TEST-day-root TEST-week-day-root"
    title="December 28, 1999"
  >
    <div
      className="TEST-day-content "
    >
      <h6
        className="TEST-day-date-label TEST-different-month"
      >
        28
      </h6>
    </div>
  </div>
  <div
    className="TEST-day-root TEST-week-day-root"
    title="December 29, 1999"
  >
    <div
      className="TEST-day-content "
    >
      <h6
        className="TEST-day-date-label TEST-different-month"
      >
        29
      </h6>
    </div>
  </div>
  <div
    className="TEST-day-root TEST-week-day-root"
    title="December 30, 1999"
  >
    <div
      className="TEST-day-content "
    >
      <h6
        className="TEST-day-date-label TEST-different-month"
      >
        30
      </h6>
    </div>
  </div>
  <div
    className="TEST-day-root TEST-week-day-root"
    title="December 31, 1999"
  >
    <div
      className="TEST-day-content "
    >
      <h6
        className="TEST-day-date-label TEST-different-month"
      >
        31
      </h6>
    </div>
  </div>
  <div
    className="TEST-day-root TEST-week-day-root"
    title="January 1, 2000"
  >
    <div
      className="TEST-day-content "
    >
      <h6
        className="TEST-day-date-label TEST-different-month"
      >
        1
      </h6>
    </div>
  </div>
</div>
`;

exports[`Week component Week with Day component 1`] = `
<div
  className="TEST-week-root "
  data-component-id="week"
>
  day comp
  day comp
  day comp
  day comp
  day comp
  day comp
  day comp
</div>
`;

exports[`Week component Week with full props 1`] = `
<div
  className="TEST-week-root rcn"
  data-component-id="week"
>
  day comp
  day comp
  day comp
  day comp
  day comp
  day comp
  day comp
</div>
`;
